tests/integration/devfile/cmd_devfile_storage_test.go

```diff
@@ -383,4 +389,43 @@ var _ = Describe("odo devfile storage command tests", func() {
 			})
 		})
 	})
+
+	When("creating a nodejs component with devfile v2.1.0 and two storages, one persistent and one ephemeral", func() {
+
+		BeforeEach(func() {
+			helper.Cmd("odo", "create", cmpName, "--context", commonVar.Context, "--project", commonVar.Project, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-2.1.0.yaml")).ShouldPass()
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+
+			helper.Cmd("odo", "storage", "create", "ephemeral-storage", "--ephemeral", "--context", commonVar.Context).ShouldPass()
+			helper.Cmd("odo", "storage", "create", "persistent-storage", "--context", commonVar.Context).ShouldPass()
+		})
+
+		When("ephemeral is set to true in preference.yaml and run odo push", func() {
+			BeforeEach(func() {
+				helper.Cmd("odo", "preference", "set", "ephemeral", "true").ShouldPass()
+				helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+			})
+
+			It("should create only one PVC", func() {
+				pvcs := commonVar.CliRunner.GetAllPVCNames(commonVar.Project)
+				Expect(len(pvcs)).To(Equal(1))
+				Expect(pvcs[0]).To(ContainSubstring("persistent-storage"))
+			})
+		})
```

Comment:
As I understand, the Ephemeral preference is only the OdoSource storage (the one that is created internally dy odo, but not referenced annywhere in the devfile). For other volumes referenced in the devfile, this will be the spec of the volume that will indicate if the volume should be persistent or ephemeral.